### ISSUE TITLE
zkgroups.so on non-x86 platforms

### ISSUE THREAD

**Issue by thefinn93** [Turasa/libsignal-service-java#24]
Hello! Thanks for all your work on libsignal and signal-cli. I'm been seeing user reports of issues loading `zkgroups.so` on non-x86 platforms. I see signal-cli got a few issues about this as well. Have you looked into fixing this at all? I saw that there is a temporary workaround in place for now.

**Comment by AsamK** [comment]
Hi, yeah I been seeing these issues as well. I've added the workaround to keep signal-cli working on those platforms, as long as signal-cli doesn't support v2 groups, this is good enough for me.
I think to really resolve the issue the zkgroups library would need to be compiled for each platform separately. Another possibility would be to compile zkgroups to webassembly and run that using wasmer or something similar ...

**Comment by Atomic-Bean** [comment]
Sorry to bug you, but do you know the steps for compiling zkgroups.so for our own architecture/s? "signalapp" don't provide any instructions so I tried `./gradlew build`, but it complains about android dependencies. The closest I can see is [this](https://github.com/signalapp/zkgroup/actions/runs/89682572/workflow) but I don't know how to run that.

**Comment by thefinn93** [comment]
in the `arm-builds` of the signald branch there has been some progress. We have a CI that automatically builds the `.so` and inserts it into the jar. Haven't done a whole lot of testing, which is why it's currently unmerged. [here is the CI file that does it](https://gitlab.com/thefinn93/signald/-/blob/arm-builds/.gitlab-ci.yml).

**Comment by Atomic-Bean** [comment]
> in the `arm-builds` branch of the signald there has been some progress. We have a CI that automatically builds the `.so` and inserts it into the jar. Haven't done a whole lot of testing, which is why it's currently unmerged. [here is the CI file that does it](https://gitlab.com/thefinn93/signald/-/blob/arm-builds/.gitlab-ci.yml).

Thanks, but leaves me more confused. Is `signald` an alternative to `signal-cli` or does `signal-cli` depend on `signald` for its daemon/dbus stuff? I'm rather new to this so I'm struggling to understand what I need to do/get to make v2 groups work.

**Comment by thefinn93** [comment]
Ah, sorry, signald is another project that depends on libsignal, it's technically a fork of signal-cli but has slightly different goals. You should be able to use our zkgroup build and jar update procedure on signal-cli with no major changes.

**Comment by Atomic-Bean** [comment]
I've tried running the commands for the `java11 aarch64` platform (Since I think that'll work for all java11 64-bit systems). I cloned the same version linked above and ran `make installDist` successfully, but `tools/repackage-zkgroup.sh zkgroup/target/release/libzkgroup.so "0.7.0"` says `No prebuilt` which seems to be because `zkgroup/target/release/libzkgroup.so` doesn't exist. I've never used ci's before so I'm probably missing an obvious step.

But also, once I have the compiled library, how do I tell programs like signal-cli where it is? Do I need to put the file somewhere special or do I modify the calling command to tell the program where it is?
